### PR TITLE
Fix macOS CI

### PR DESCRIPTION
**Pull request by BenWibking**
## PR Summary

This fixes the GitHub macOS CI and adds a root-level `requirements.txt` file for Python package dependencies.

## PR Checklist

- [ ] Code passes cpplint
- [ ] New features are documented.
- [ ] Adds a test for any bugs fixed. Adds tests for new features.
- [ ] Code is formatted
- [ ] Changes are summarized in CHANGELOG.md
- [ ] Change is breaking (API, behavior, ...)
  - [ ] Change is *additionally* added to CHANGELOG.md in the breaking section
  - [ ] PR is marked as breaking
  - [ ] Short summary API changes at the top of the PR (plus optionally with an automated update/fix script)
- [ ] CI has been triggered on [Darwin](https://re-git.lanl.gov/eap-oss/parthenon/-/pipelines) for performance regression tests.
- [ ] Docs build
- [ ] (@lanl.gov employees) Update copyright on changed files

### CODE REVIEW

**Comment by BenWibking** [comment]
@Yurlungur Any ideas on why the macOS CI isn't triggered? 😕 

Edit: nevermind, figured it out: https://github.com/parthenon-hpc-lab/parthenon/actions/workflows/ci-macos.yml

**Comment by BenWibking** [comment]
hmm, we might need to add a requirements.txt file for the python modules:
```
Error: No file in /Users/runner/work/parthenon/parthenon matched to [**/requirements.txt or **/pyproject.toml], make sure you have checked out the target repository
```

**Comment by Yurlungur** [comment]
> hmm, we might need to add a requirements.txt file for the python modules:
> 
> ```
> Error: No file in /Users/runner/work/parthenon/parthenon matched to [**/requirements.txt or **/pyproject.toml], make sure you have checked out the target repository
> ```

that's probably worth doing anyway. ;)

**Comment by BenWibking** [comment]
~~If the full test suite takes too long on macOS, I will switch to running `ctest -LE mpi` instead.~~

File: .github/workflows/ci-macos.yml [hunk 1]
```diff
@@ -11,7 +11,7 @@ concurrency:
 env:
   CTEST_OUTPUT_ON_FAILURE: 1
   CMAKE_BUILD_PARALLEL_LEVEL: 4 # num threads for build
-
+
```
Comment (pdmullen) [inline .github/workflows/ci-macos.yml]: Can you adjust your text editor settings to not add empty whitespace here?

File: .github/workflows/ci-macos.yml [hunk 2]
```diff
@@ -24,23 +24,27 @@ jobs:
...
       with:
         python-version: '3.11'
-        cache: 'pip' # caching pip dependencies
+        cache: 'pip'
+        cache-dependency-path: '**/requirements.txt'
+    - run: pip install -r requirements.txt
+
+    - name: Install dependencies
+      run: |
+        brew install openmpi hdf5-mpi adios2 || true
```
Comment (pdmullen) [inline .github/workflows/ci-macos.yml:33]: How much total time is added from these new macos tests?

File: .github/workflows/ci-macos.yml [hunk 3]
```diff
@@ -24,23 +24,27 @@ jobs:
     - uses: actions/setup-python@v5
       with:
         python-version: '3.11'
-        cache: 'pip' # caching pip dependencies
+        cache: 'pip'
+        cache-dependency-path: '**/requirements.txt'
```
Comment (pdmullen) [inline .github/workflows/ci-macos.yml:28]: Do you have to provide this auxillary file? Or can the three dependencies just be listed in this file somehow?

**Review by pdmullen — APPROVED** [review]

File: .github/workflows/ci-macos.yml (hunk 3 above)
Comment (BenWibking) [reply .github/workflows/ci-macos.yml:28]: It has to be provided in order to cache the dependencies (described here): https://github.com/actions/setup-python/tree/main?tab=readme-ov-file#caching-packages-dependencies

File: .github/workflows/ci-macos.yml (hunk 2 above)
Comment (BenWibking) [reply .github/workflows/ci-macos.yml:33]: It takes 8 minutes to run (https://github.com/parthenon-hpc-lab/parthenon/actions/runs/9196810177) and runs in parallel with the other CI actions that take significantly longer, so it doesn't add any time to the overall checks.

**Review by Yurlungur — APPROVED** [review]

**Pull request merged**